server/repository/src/db_diesel/stocktake_line.rs

```diff
@@ -175,3 +196,25 @@ fn create_filtered_query(filter: Option<StocktakeLineFilter>) -> BoxedStocktakeL
 
     query
 }
+
+fn apply_item_filter(
+    query: BoxedStocktakeLineQuery,
+    filter: Option<StocktakeLineFilter>,
+    connection: &StorageConnection,
+    store_id: String,
+) -> BoxedStocktakeLineQuery {
+    if let Some(f) = filter {
+        if let Some(item_code_or_name) = &f.item_code_or_name {
+            let mut item_filter = ItemFilter::new();
+            item_filter.code_or_name = Some(item_code_or_name.clone());
+            item_filter.is_visible = Some(true);
+            let items = ItemRepository::new(connection)
+                .query_by_filter(item_filter, Some(store_id))
+                .unwrap();
```

Comment:
Is this unwrap safe? If so can you comment why?

Reply:
I copied it from the stock_line filter, and figured that if it's good enough for them, then it's good enough for me.